lib/collection/certificate.js

```diff
@@ -0,0 +1,129 @@
+* @property {String} [name] A name for the certificate
+* @property {Array} [matches] A list of match patterns
+* @property {String} [keyPath] The path on the file system to the file that contains the key
+* @property {String} [pemPath] The path on the file system to the certificate file
+* @property {String} [passphrase] The passphrase for the certificate key
+*
+* @example <caption>JSON definition of an example certificate object</caption>
+* {
+*     name: 'My certificate for example.com',
+*     matches: ['https://example.com/*'],
+*     key: { src: '/User/path/to/certificate/key' },
+*     cert: { src: '/User/path/to/certificate' },
+*     passphrase: 'iampassphrase'
+* }
+*/
+_.inherit((
+
+    /**
+     * A Certificate definition that represents the ssl certificate
+     * to be used for an url.
+     * Properties can then use the `.toObjectResolved` function to procure an object representation of the property with
+     * all the variable references replaced by corresponding values.
+     * @constructor
+     * @extends {Property}
+     *
+     * @param {Certificate~definition=} [options] Object with host, key, pem and passphrase
+     *
+     * @example <caption> Create a new Certificate</caption>
+     *
+     * var Certificate = require('postman-collection').Certificate,
+     *    certificate = new Certificate({
+     *     id: 'auto generated',
+     *     name: 'Certificate for example',
+     *     matches: ['example.com'],
+     *     key: { src: '/User/path/to/certificate/key' },
+     *     cert: { src: '/User/path/to/certificate' },
+     *     passphrase: 'iampassphrase'
+     * });
+     */
+    Certificate = function Certificate (options) {
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        Certificate.super_.apply(this, options);
+
+        // return early if options is empty or invalid
+        if (!_.isObject(options)) {
+            return;
+        }
+
+        _.mergeDefined(this, {
+            id: options.id,
+            name: options.name,
+
+            matches: options.matches && new UrlMatchPatternList({}, options.matches),
+
+            key: options.key,
+            cert: options.cert,
+            passphrase: options.passphrase
+        });
+    }), Property);
+
+_.assign(Certificate.prototype, /** @lends Certificate.prototype */ {
+    /**
+     * Ensure all object have id
+     * @private
+     */
+    _postman_requiresId: true,
+
+    /**
+     * Tests the url string to check if it matches any of the allowed match patterns.
+     *
+     * @param {String=} [urlStr] The url string for which the proxy match needs to be done.
+     */
+    test: function(url) {
+        // test the input url against allowed matches
+        return this.matches.test(url);
+    },
+
+    /**
+     * Returns the object representation of the certificate object.
+     *
+     * @return {Object}
+     */
+    toJSON: function() {
```

Comment:
[Property Base](https://github.com/kamalaknn/postman-collection/blob/37ee06cc15ca51e713c00178c87624eb8484f06e/lib/collection/property-base.js#L66) Will handle this.